tests/Bridge/Laravel/Http/Middlewares/AuditMiddlewareTest.php

```diff
@@ -0,0 +1,259 @@
+
+        self::assertSame('OK', $response->getContent());
+    }
+
+    /**
+     * Test if handle can catch the exceptions and continue processing
+     * and rethrow exception at end
+     *
+     * @return void
+     *
+     * @throws \EoneoPay\Utils\Exceptions\InvalidDateTimeStringException
+     */
+    public function testHandleHandlesExceptionsAndRethrows(): void
+    {
+        $this->expectException(InvalidDocumentClassException::class);
+        $this->expectExceptionMessage('Something failed.');
+
+        $middleware = $this->getMiddleware();
+
+        $middleware->handle(
+            $this->getRequest(),
+            static function (): void {
+                throw new InvalidDocumentClassException('Something failed.');
+            }
+        );
+    }
+
+    /**
+     * Test if handle can intercept response which is already in PSR7 format
+     *
+     * @return void
+     *
+     * @throws \EoneoPay\Utils\Exceptions\InvalidDateTimeStringException
+     */
+    public function testHandleWhenResponseFromControllerIsAlreadyPsr7(): void
+    {
+        $middleware = $this->getMiddleware();
+
+        $response = $middleware->handle(
+            $this->getRequest(),
+            static function (): ResponseInterface {
+                return new TextResponse('OK');
+            }
+        );
+
+        self::assertInstanceOf(ResponseInterface::class, $response);
+        self::assertSame('OK', (string)$response->getBody());
+    }
+
+    /**
+     * Test if exception thrown by http logger service is logged using logger interface
+     *
+     * @return void
+     *
+     * @throws \EoneoPay\Utils\Exceptions\InvalidDateTimeStringException
+     */
+    public function testHttpLoggerExceptionsAreLoggedByLoggerInterface(): void
+    {
+        $httpLogger = new HttpLoggerExceptionStub(); // this stub specifically throws an exception
+        $logHandler = new LogHandlerStub();
+        $middleware = $this->getMiddleware($httpLogger, $logHandler);
+
+        $middleware->handle(
+            $this->getRequest(),
+            static function (): Response {
+                return new Response('OK');
+            }
+        );
+
+        self::assertCount(1, $logHandler->getLogs());
+        $log = $logHandler->getLogs()[0];
+        self::assertArrayHasKey('message', $log);
+    }
+
+    /**
+     * Test to assert that an exception is caught and the process
+     * continues to generate psr7 request/response
+     *
+     * @return void
+     */
+    public function testProcessingContinuesWhenExceptionIsThrownInOriginalProcess(): void
+    {
+        $httpLogger = new HttpLoggerStub();
+        $middleware = $this->getMiddleware($httpLogger);
+
+        try {
+            $middleware->handle(
+                $this->getRequest(),
+                static function (): void {
+                    throw new InvalidDocumentClassException('Something failed.');
+                }
+            );
+        } catch (Exception $exception) {
```

Comment:
Shouldn't this be InvalidDocumentClassException ?

Reply:
this is trying to emulate the working of the middleware. As it catches all exception and not specific one's (because of what it needs to do), and I was not trying to match the exception catching here. The whole logic here was to get the test to proceed so that I can check the values were passed to HttpLogger as expected.

Reply:
Yep, I see that. You should catch InvalidDocumentClassException, because you expect that. However, if you something throws a different Exception than `InvalidDocumentClassException` you'd want to throw that in the test, because it means something else has happened which wasn't expected (in this test).

Reply:
done.